src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -407,12 +409,28 @@ public CodeFixProvider GetSuppressionFixer(string language, IEnumerable<string>
             foreach (var fixer in allFixers)
             {
                 await extensionManager.PerformActionAsync(fixer, () => fixer.RegisterCodeFixesAsync(context) ?? SpecializedTasks.EmptyTask).ConfigureAwait(false);
-                if (!fixes.Any())
+                foreach (var fix in fixes)
                 {
-                    continue;
-                }
+                    if (!fix.Action.PerformFinalApplicabilityCheck)
+                    {
+                        return true;
+                    }
 
-                return true;
+                    // Have to see if this fix is still applicable.  Jump to the foreground thread
+                    // to make that check.
```

Comment:
Discussed offline with @davkean  and @jasonmalinowski .  This is the pattern that Roslyn currently uses.  We never use JTF.  AFAICT, we try to make sure we never pump.  Instead, we have a strict rule that if UI thread is blocking on the results of a background call, that that background call cannot wait on the UI thread.

In this case, the codepath that runs through this never does any blocking on the UI thread.  As such, it's safe for the BG thread to make a BG-FG-BG switch.